### PR TITLE
Added brand filters and homepage category links

### PR DESCRIPTION
**Pull request by shrushti2000**
### 1. Added links to categories on Homepage:- 
When the user clicks on any category on homepage he will  be redirected to product listing page and will be  shown products belonging to that category .(eg. makeup, skincare,haircare etc...)
ezgif com-gif-maker (19)

### 2. Added  links to sub categories on homepage:-
When user clicks on any sub category present in the subheader on homepage he will be  redirected to product listing page and will be  shown products belonging to that sub category .(eg. lipsticks, sunscreen,mascara etc..)
ezgif com-gif-maker (18)

### 3. Added filter by brand on product listing page.
ezgif com-gif-maker (17)

### CODE REVIEW

File: src/pages/HomePage/HomePage.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ import Carousel from '../../components/Carousel/Carousel'
 import brandImage1 from '../../assets/Images/loreal_logo.jpg'
 import brandImage2 from '../../assets/Images/lakme_logo.jpg'
 import brandImage4 from '../../assets/Images/maybelline_logo.png'
-import brandImage5 from '../../assets/Images/Wishper_logo.png'
+import brandImage5 from '../../assets/Images/mamaearth-logo.jpg'
+import brandImage6 from '../../assets/Images/wow-logo.png'
+import brandImage7 from '../../assets/Images/Wishper_logo.png'
 import offerImage1 from '../../assets/Images/offerImage1.jpg'
 import offerImage2 from '../../assets/Images/offerImage2.jpg'
```
Comment (sharathnair9999) [inline src/pages/HomePage/HomePage.js:11]: you can keep all imports in one file and take it from there. readability wont get hampered by doing that

File: src/pages/HomePage/HomePage.js [hunk 2]
```diff
@@ -1,71 +1,58 @@
...
+        <Link to="/productlistingpage"><img className='brand-img' src={brandImage2} alt="brand-img" onClick={(e) => dispatch({ type: 'SET_BRAND', payload: "Lakme" }, dispatch({ type: 'SET_SUB_CATEGORY', payload: null }))} /></Link>
+        <Link to="/productlistingpage"><img className='brand-img' src={brandImage4} alt="brand-img" onClick={(e) => dispatch({ type: 'SET_BRAND', payload: "Maybelline" }, dispatch({ type: 'SET_SUB_CATEGORY', payload: null }))} /></Link>
+        <Link to="/productlistingpage"><img className='brand-img' src={brandImage7} alt="brand-img" onClick={(e) => dispatch({ type: 'SET_BRAND', payload: "Whisper" }, dispatch({ type: 'SET_SUB_CATEGORY', payload: null }))}/></Link>
+        <Link to="/productlistingpage"><img className='brand-img' src={brandImage5} alt="brand-img" onClick={(e) => dispatch({ type: 'SET_BRAND', payload: "Mamaearth" }, dispatch({ type: 'SET_SUB_CATEGORY', payload: null }))} /></Link>
+        <Link to="/productlistingpage"><img className='brand-img' src={brandImage6} alt="brand-img" onClick={(e) => dispatch({ type: 'SET_BRAND', payload: "Wow" }, dispatch({ type: 'SET_SUB_CATEGORY', payload: null }))}/></Link>
+
```
Comment (dev-enforced) [inline src/pages/HomePage/HomePage.js:53]: You can make an array of FeaturedBrands having the imageUrl and respective payload values which you can map over and add the links here

File: src/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -1,71 +1,58 @@
...
+import brandImage6 from '../../assets/Images/wow-logo.png'
+import brandImage7 from '../../assets/Images/Wishper_logo.png'
 import offerImage1 from '../../assets/Images/offerImage1.jpg'
 import offerImage2 from '../../assets/Images/offerImage2.jpg'
 import { Link } from 'react-router-dom'
 import './HomePage.css'
 import Footer from '../../components/Footer/Footer'
+import { useContext } from 'react'
+import { StateContext } from '../../Context'
+import { useEffect } from 'react'
```
Comment (dev-enforced) [inline src/pages/HomePage/HomePage.js]: You can write the 2 modules of useContext and useEffect together too `import {useContext,useEffect} from "react"`

File: src/utilityFunctions.js [hunk 4]
```diff
@@ -26,11 +26,22 @@ export const getRatingSortedData = (data, rating) => {
...
     console.log('filter data')
     let filteredData = [...data];
     if (state.sortByFilters.sortByCategory.length !== 0) {
         filteredData = filteredData.filter(item => state.sortByFilters.sortByCategory.includes(item.categoryName))
     }
+    if (state.sortByFilters.sortByBrands.length !== 0) {
+        filteredData = filteredData.filter(item => state.sortByFilters.sortByBrands.includes(item.brand))
+    }
+    // if (state.sortByFilters.sortBySubCategory.length !== 0) {
+    //     filteredData = filteredData.filter(item => state.sortByFilters.sortBySubCategory.includes(item.subCategoryName))
```
Comment (dev-enforced) [inline src/utilityFunctions.js:39]: You can remove this commented line of code if you don't want to use it further

**Review by dev-enforced — APPROVED** [review]
The code looks very well structured and the PR description also made clear about the features present in it. The features are working very well Shrushti. Good work!! Passed some minor suggestions which you can add on later.

**Comment by shrushti2000** [comment]
Thank you @sharathnair9999 @dev-enforced for PR review. Will surely work upon the suggestions.

**Pull request merged**